### PR TITLE
Increase tests coverage

### PR DESCRIPTION
**Pull request by darkdarkdragon**

### CODE REVIEW

File: src/api/common/schemas/pathfind.json [hunk 1]
```diff
@@ -24,7 +24,32 @@
...
+            },
+            "currency": {
+              "description": "The three-character code or hex string used to denote currencies",
+              "$ref": "currency"
+            },
+            "counterparty": {
+              "description": "The Ripple account address of the currency's issuer or gateway",
+              "$ref": "address"
+            },
+            "required": ["currency", "value"]
```
Comment (clark800) [inline src/api/common/schemas/pathfind.json]: So this change is to make counterparty optional? Is there documentation to confirm that the destination counterparty is optional for non-XRP?

File: test/api-test.js [hunk 2]
```diff
@@ -224,8 +235,10 @@ describe('RippleAPI', function() {
 
   it('getOrderbook - direction is correct for bids and asks', function(done) {
     this.api.getOrderbook(address, orderbook, {}, (error, data) => {
-      assert(_.every(data.bids, bid => bid.specification.direction === 'buy'));
-      assert(_.every(data.asks, ask => ask.specification.direction === 'sell'));
+      assert.ok(
```
Comment (clark800) [inline test/api-test.js]: What is the `.ok` for?

File: test/api-test.js [hunk 3]
```diff
@@ -262,7 +275,134 @@ describe('RippleAPI', function() {
...
-      _.partial(checkResult, responses.getPaths, done));
+      _.partial(checkResult, responses.getPaths.XrpToUsd, done));
+  });
+
+  // @TODO
+  // need decide what to do with currencies/XRP:
+  // if add 'XRP' in currencies, then there will be exception in
+  // xrpToDrops function (called from toRippledAmount)
+  it('getPaths USD 2 USD', function(done) {
+    const pathfind = {
```
Comment (clark800) [inline test/api-test.js]: Maybe we should put these request objects in the fixtures?

**Comment by darkdarkdragon** [comment]
@wltsmrz Turns  out I've got old version of assert-diff installed

File: src/api/common/schemas/pathfind.json (hunk 1 above)
Comment (darkdarkdragon) [inline src/api/common/schemas/pathfind.json]: Yes, that make it optional.
Doc is [here](https://github.com/ripple/ripple-lib/blob/develop/src/api/ledger/pathfind.js#L37)

Comment (clark800) [inline src/api/common/schemas/pathfind.json]: Ok thanks

File: test/api-test.js (hunk 2 above)
Comment (darkdarkdragon) [inline test/api-test.js]: Same as just assert().
Turned out that I has old assert-diff installed where just assert() doesn't work.
I will change assert.ok back to normal in next commit.

File: test/api-test.js (hunk 3 above)
Comment (darkdarkdragon) [inline test/api-test.js]: Yes, good point.

**Comment by darkdarkdragon** [comment]
@clark800 done fixes

File: test/api-test.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ const hashes = require('./fixtures/hashes');
 const MockPRNG = require('./mock-prng');
 const sjcl = require('../src').sjcl;
 const address = addresses.ACCOUNT;
+const NotFoundError = require('../src/api/common/errors').NotFoundError;
```
Comment (clark800) [inline test/api-test.js]: Should use `this.api.errors.NotFoundError` to use the real interface

File: test/api-test.js [hunk 5]
```diff
@@ -225,7 +236,8 @@ describe('RippleAPI', function() {
   it('getOrderbook - direction is correct for bids and asks', function(done) {
     this.api.getOrderbook(address, orderbook, {}, (error, data) => {
       assert(_.every(data.bids, bid => bid.specification.direction === 'buy'));
-      assert(_.every(data.asks, ask => ask.specification.direction === 'sell'));
+      assert(
```
Comment (clark800) [inline test/api-test.js]: Unnecessary change?

Comment (darkdarkdragon) [inline test/api-test.js]: too long line

File: test/api-test.js (hunk 4 above)
Comment (darkdarkdragon) [inline test/api-test.js]: @clark800  Fixed

**Comment by clark800** [comment]
LGTM

**Comment by clark800** [comment]
Can you squash to get rid of the commit about assert.ok?

**Comment by darkdarkdragon** [comment]
yep
